cmd/role/role.go

```diff
@@ -0,0 +1,377 @@
+		}
+		authApi.GetInfo("", "")
+
+		roleName, _ := cmd.Flags().GetString("role-name")
+
+		permissionsMap := map[string][]string{}
+
+		permissionsList, _ := cmd.Flags().GetStringArray("permissions")
+		for _, permissionsString := range permissionsList {
+			permission := strings.Split(permissionsString, ",")
+
+			if len(permission) != 2 {
+				logrus.Fatalln("All necessary fields not provided for permissions")
+			}
+
+			kvpOne := strings.Split(permission[0], "=")
+			if len(kvpOne) != 2 {
+				logrus.Fatalln("Incorrect format in permissions fields")
+			}
+
+			kvpTwo := strings.Split(permission[1], "=")
+			if len(kvpTwo) != 2 {
+				logrus.Fatalln("Incorrect format in permissions fields")
+			}
+
+			if kvpOne[0] == "resource-type" && kvpTwo[0] == "operation-group" {
+				CreatePermissionsMap(kvpOne[1], kvpTwo[1], permissionsMap)
+			} else if kvpTwo[0] == "resource-type" && kvpOne[0] == "operation-group" {
+				CreatePermissionsMap(kvpTwo[1], kvpOne[1], permissionsMap)
+			} else {
+				logrus.Fatalln("Resource type and Operation group must be specified in permissions")
+			}
+		}
+
+		roleSpec, err := authApi.CreateRoleSpec(cmd, roleName, permissionsMap)
+		if err != nil {
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		resp, r, err := authApi.CreateRole().RoleSpec(*roleSpec).Execute()
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		if viper.GetString("output") == "table" {
+			fullRoleContext := *formatter.NewFullRoleContext()
+			fullRoleContext.Output = os.Stdout
+			fullRoleContext.Format = formatter.NewFullRoleFormat(viper.GetString("output"))
+			fullRoleContext.SetFullRole(resp.GetData())
+			fullRoleContext.Write()
+			return
+		}
+
+		rolesCtx := formatter.Context{
+			Output: os.Stdout,
+			Format: formatter.NewFullRoleFormat(viper.GetString("output")),
+		}
+
+		formatter.SingleRoleWrite(rolesCtx, resp.GetData())
+	},
+}
+
+func CreatePermissionsMap(resourceType string, operationGroup string, permissionsMap map[string][]string) {
+	if ops, ok := permissionsMap[resourceType]; ok {
+		ops = append(ops, operationGroup)
+		permissionsMap[resourceType] = ops
+	} else {
+		permissionsMap[resourceType] = []string{operationGroup}
+	}
+}
+
+var updateRoleCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Update a custom role",
+	Long:  "Update a custom role in YugabyteDB Managed",
+	Run: func(cmd *cobra.Command, args []string) {
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf("could not initiate api client: %s", err.Error())
+		}
+		authApi.GetInfo("", "")
+
+		roleName, _ := cmd.Flags().GetString("role-name")
+		roleId, err := authApi.GetRoleIdByName(roleName)
+		if err != nil {
+			logrus.Fatal(err)
+		}
+
+		permissionsMap := map[string][]string{}
+
+		permissionsList, _ := cmd.Flags().GetStringArray("permissions")
+		for _, permissionsString := range permissionsList {
```

Comment:
This section seems to be common between create/update of role? Any chance we can reuse the code?